File: sphone-api/Domain/Customers/Events/CustomerRegisteredEvent.cs
using SPhone.Domain.Common;

namespace SPhone.Domain.Customers.Events;

public sealed record CustomerRegisteredEvent(
    Guid EventId,
    DateTime OccurredAt,
    Guid CustomerId,
    string NationalId,
    string FullName,
    string PhoneNumber) : IDomainEvent
{
    public static CustomerRegisteredEvent Create(CustomerId customerId, NationalId nationalId, string fullName, PhoneNumber phoneNumber) =>
        new(Guid.NewGuid(), DateTime.UtcNow, customerId.Value, nationalId.Value, fullName, phoneNumber.Value);
}

File: sphone-api/src/domain/customers/events/CustomerRegisteredEvent.ts
import { DomainEvent } from '../../shared/DomainEvent'
import { v4 as uuidv4 } from 'uuid'

export class CustomerRegisteredEvent implements DomainEvent {
  readonly eventId: string
  readonly occurredAt: Date
  readonly eventType = 'CustomerRegistered'

  constructor(
    public readonly customerId: string,
    public readonly nationalId: string,
    public readonly fullName: string,
  ) {
    this.eventId = uuidv4()
    this.occurredAt = new Date()
  }
}
